test/common/grpc/codec_test.cc

```diff
@@ -221,6 +221,68 @@ TEST(GrpcCodecTest, decodeMultipleFrame) {
   }
 }
 
+TEST(GrpcCodecTest, decodeSingleFrameOverLimit) {
+  helloworld::HelloRequest request;
+  std::string test_str = std::string(64 * 1024, 'a');
+  request.set_name(test_str);
+
+  Buffer::OwnedImpl buffer;
+  std::array<uint8_t, 5> header;
+  Encoder encoder;
+  encoder.newFrame(GRPC_FH_DEFAULT, request.ByteSize(), header);
+  buffer.add(header.data(), 5);
+  buffer.add(request.SerializeAsString());
+  size_t size = buffer.length();
+
+  std::vector<Frame> frames;
+  // Configure decoder with 32kb max_frame_length.
+  Decoder decoder;
+  decoder.setMaxFrameLength(32 * 1024);
+
+  // The decoder doesn't successfully decode due to oversized frame.
+  EXPECT_FALSE(decoder.decode(buffer, frames));
+  EXPECT_EQ(buffer.length(), size);
+}
+
+TEST(GrpcCodecTest, decodeMultipleFramesOverLimit) {
+  Buffer::OwnedImpl buffer;
+  std::array<uint8_t, 5> header;
+  Encoder encoder;
+
+  // First frame is valid (i.e. within max_frame_length limit).
+  helloworld::HelloRequest request;
+  request.set_name("hello");
+  encoder.newFrame(GRPC_FH_DEFAULT, request.ByteSize(), header);
+  buffer.add(header.data(), 5);
+  buffer.add(request.SerializeAsString());
+
+  // Second frame is invalid (i.e. exceeds max_frame_length).
+  helloworld::HelloRequest overlimit_request;
+  std::string test_str = std::string(64 * 1024, 'a');
```

Comment:
A more subtle test would have two frames, both below the limit but cummulatively over the limit.

Reply:
Good Idea!

I think you meant single frame with multiple buffers, each of which is below limit but cumulatively over frame limit. I have added test `decodeSingleFrameWithMultiBuffersOverLimit` above. 
PTAL, Thanks!